src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,345 @@
+                        $component->setTypeGraph('bar');
+                    }
+                    //.......Save changes made on component
+                    $em->persist($component);
+                    $em->flush();
+
+                    return $this->redirectToRoute('app_dashboard', array(
+                        'id' => $id,
+                        'component' => $component,
+                    ));
+                }
+            }
+        }
+
+        //.......If dashboard edit form was submitted, save changes to database
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em->persist($dashboard);
+            $em->flush();
+
+            return $this->redirectToRoute('app_dashboard', array(
+                'id' => $id));
+        }
+        //.......Create the view for each component form
+        foreach ($forms as $key => $formOne) {
+            $forms[$key] = $formOne->createView();
+        }
+        //.......Render dashboard view and send the variables to it
+        return $this->render('AppBundle:App:dashboard.html.twig', array(
+            'userId' => $userId,
+            'user' => $user,
+            'creator' => $creator,
+            'allCharts' => $charts,
+            'componentName' => $componentName,
+            'componentId' => $componentId,
+            'dashboard' => $dashboard,
+            'components' => $components,
+            'id' => $id,
+            'form' => $form->createView(),
+            'forms' => $forms,
+            'size' => $sizeComponent
+        ));
+    }
+
+    //.......Share Dashboard Function
+    /***
+     * @param Request $request
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function shareAction(Request $request, $id)
+    {
+        //.......Get current dashboard
+        $em = $this->getDoctrine()->getManager();
+        $dashboard = $em
+            ->getRepository('AppBundle:Dashboard')
+            ->find($id);
+        //.......Get the parameter from url for search function
+        $parameter = $request->get('search');
+        //.......Get all the user that have in the last name the "value" parameter
+        $listUsers = $em
+            ->getRepository('AppBundle:User')
+            ->createQueryBuilder('user')
+            ->where('user.lastName LIKE :lastName')
+            ->setParameter('lastName', '%' . $parameter . '%')
+            ->getQuery()
+            ->getResult();
+
+        //.......if form has been submitted removes all collaborators from dashboard,
+        // ......adds the checked ones to the dashboard and saves them in the database
+        if ($request->get('activeUsers')) {
+            $activeUsers = $request->get('activeUsers');
+
+            foreach ($listUsers as $inactiveUser) {
+                $dashboard->removeCollaborator($inactiveUser);
+                $em->persist($dashboard);
```

Comment:
You don't need to persist the dashboard again, it already exists. Just flush changes after the loops